ws/ws.nim

```diff
@@ -250,17 +313,23 @@ proc handleRequest*(
     raise newException(WSHandshakeError,
         "Failed to sent handshake response. Error: " & exc.msg)
 
-  return WSSession(
+  let session = WSSession(
     readyState: ReadyState.Open,
     stream: request.stream,
     proto: protocol,
+    extensions: system.move(ws.extensions),
```

Comment:
I think `extensions` is shared across all sessions in server mode?

Reply:
I guess I'm a bit confused about the `move` here, why do we need it?

Reply:
no, it is not shared. the server will create new one for each session.

Reply:
for extension such as `permessage-deflate`, as it's name implies. will keep a de/compression context from the start of message to the end of message. from first frame to last frame will use the same context. so extensions cannot be shared among session.

Reply:
for a simple extension without configuration parameter, may the factory can keep a singleton to be shared among session.
but again, the extension chain itself can be different from one session to the next.

Reply:
> for extension such as permessage-deflate, as it's name implies. will keep a de/compression context from the start of message to the end of message. from first frame to last frame will use the same context. so extensions cannot be shared among session.

OK right, that makes sense then.